config/environments/production.rb

```diff
@@ -89,6 +89,7 @@
       h[:request_id] = event.payload[:request_id]
       h[:requested_by] = event.payload[:requested_by] if event.payload[:requested_by]
       h[:form_id] = event.payload[:form_id] if event.payload[:form_id]
+      h[:query] = event.payload[:path].split("?").second
```

Comment:
This will work but parsing the params into a json object/hash of key values will make using them in Splunk in future simpler (I'd hope that the logging formatter accepts objects for the values and converts appropriately but I cannot be sure without testing it). I have it in mind to tweak our Splunk set up so that by default it parses the json format of the app logs into json and displays it neatly (this makes searching and extracting values much simpler), so it'll look like this (I've added in the query params just to demonstrate:

![image](https://github.com/alphagov/forms-api/assets/31340341/087f29c5-176e-488d-8e72-d44094d492e8)

Then it'll be simple to query such as:
```
| spath output=organisation_id query.organisation_id
```

If its kept as a string then we'd need to `rex` it out, something like
```
rex field=_raw "organisation_id=(?<organisation_id>.*?)[&$]"
```

Reply:
Just looking at the Lograge docs they mention that the event payload includes `params` which is the hash of params (its in the what it doesn't do section but in the sense of, if you want it to then you can do it yourself): https://github.com/roidrage/lograge#what-it-doesnt-do

>The payload does already contain the params hash, so you can easily add it in manually using custom_options:
```
params: event.payload[:params].except(*exceptions)
```

Might be helpful?

Reply:
Oh nice, I did see this in the docs but wasn't sure if Splunk would handle the nested JSON okay. I like this a lot, I'll do this.

Reply:
I've updated the commit to do this 👍

Reply:
Great stuff, I think its worth deploying this into dev and we can confirm the log format all the way through into Splunk. I'm happy to help check it out once there are some logs?

Reply:
Looks good to me
```
{ [-]
   action: index
   controller: Api::V1::FormsController
   db: 1850.7
   duration: 2713.86
   format: json
   host: api.dev.forms.service.gov.uk
   method: GET
   params: { [-]
     org: government-digital-service
   }
   path: /api/v1/forms
   request_id: 56623f7b-c6ec-4a4a-b823-a987ddcaf57a
   requested_by: forms-admin-133
   status: 200
   view: 0.17
}
```

also works without params in request

```
{ [-]
   action: ping
   controller: HeartbeatController
   db: 0
   duration: 0.33
   format: html
   host: 10.10.5.122
   method: GET
   params: { [-]
   }
   path: /ping
   request_id: fcf62340-8188-434a-bfbc-1fbad8132c96
   status: 200
   view: 0.07
}
```